refarch-integrations/refarch-s3-integration/refarch-s3-integration-client/src/main/java/de/muenchen/refarch/s3/integration/client/service/S3StorageUrlProvider.java

```diff
@@ -0,0 +1,64 @@
+package de.muenchen.refarch.s3.integration.client.service;
+
+import de.muenchen.refarch.s3.integration.client.exception.PropertyNotSetException;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Provides the S3 storage URL either configured in a process configuration of a given process
+ * definition or the default URL.
+ */
+@Slf4j
+public class S3StorageUrlProvider {
+
+    private final S3DomainProvider s3DomainProvider;
+
+    @Getter
+    private final String defaultDocumentStorageUrl;
+
+    /**
+     * Constructor for client users which obtain their own {@link S3DomainProvider}.
+     *
+     * @param s3DomainProvider The {@link S3DomainProvider} to use for fetching domain-specific S3 URLs.
+     * @param defaultDocumentStorageUrl The default S3 storage URL to use if no domain-specific URL is
+     *            available.
+     */
+    public S3StorageUrlProvider(final S3DomainProvider s3DomainProvider, final String defaultDocumentStorageUrl) {
+        this.s3DomainProvider = s3DomainProvider;
+        this.defaultDocumentStorageUrl = defaultDocumentStorageUrl;
+    }
```

Comment:
Can this constructor maybe be generated via Lombok? Its a general thing you could check in all files. :) I won't comment those things further.

Reply:
Think that is more for documentation purposes as no java doc can be added to lombok constructors.
Would leave it as it is

Reply:
Ok that's a valid point.